enhancements/olm/permissions-validation-preflight-check.md

```diff
@@ -0,0 +1,455 @@
+
+The `Metadata` section above is intended to support the creation of tooling
+around the enhancement process.
+
+## Summary
+
+Validating of ServiceAccount permissions before attempting to install/manage extension content.  
+
+Instead of hoping users are able to patch together a functioning ServiceAccount, OLMv1 should perform a preflight check that tells users about any required permissions that are missing from the provided ServiceAccount. The output of this preflight check could then be utilized in future improvements to the user experience, such as:
+- Surfacing all the required permissions through a CLI tool or in GUIs
+- Automating the creation of a suggested minimally functional provided ServiceAccount
+
+
+## Motivation
+
+Installing a ClusterExtension in OLMv1 requires users to provide a ServiceAccount for installing and managing that extension. Determining the required permissions for that ServiceAccount is currently a non-trivial, repetitive, lengthy process.
+
+### User Stories
+
+- As a cluster user, I wish to be able to easily assess sufficiency of the required RBAC necessary for a `ServiceAccount` associated with a `ClusterExtension` installation attempt.
+- As a cluster user, I wish to be able to clearly assess permissions gaps for such an installation attempt so I may easily adjust them.
+- As a gitops-style actor on the cluster, I wish to be able to interpret permissions gaps programmatically so that I can handle remediation/alerts appropriately.
+
+
+### Goals
+
+Introduce a preflight check that evaluates the permissions granted to the provided ServiceAccount against the permissions required by the ClusterExtension bundle. If any permissions are missing, the system will:
+- Provide a detailed, user-friendly error message listing missing permissions.
+- Prevent installation or upgrades from proceeding until the issue is resolved.
+- The pre-flight check implemented as part of this work should include more than just surfacing the RBAC in the bundle. Generally, it should surface any permissions that are needed to stamp out all resources in the bundle on the cluster.
+- Any bundle introspection is about what needs to be applied to the cluster by OLM. OLM doesn't, and shouldn't, care about what permissions the cluster extension needs to operate as expected in the cluster. 
+
+
+### Non-Goals
+
+- Only what is necessary to install the ClusterExtension based on what is provided in the bundle should be verified. OLM will only have knowledge of Kubernetes and applying Kubernetes resources so we won't be able to verify something like access to an external secret store as part of an extension's installation. 
+- Optionality of the preflight check.
+
+## Proposal
+
+The preflight check will extend the existing preflight interface in the operator controller. Key components:
+1. Permission Analysis
+   - Use the ServiceAccount token to verify access against required permissions declared in the ClusterExtension’s bundle.
```

Comment:
I don't think we're actually planning to use the SA token in this case. Rather, we are building a local cache of all RBAC and doing a local best-effort evaluation.

Reply:
[f3a38b2](https://github.com/openshift/enhancements/pull/1768/commits/f3a38b29e6d35d65f1736ef9750a3e9073dc1c06)